### PR TITLE
fix(GcloudTokenUpdateThread): set expirity time to it's max and add timeout on token waiting

### PR DESCRIPTION
**Pull request by dkropachev**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Tested - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-2/54

File: sdcm/cluster_k8s/gke.py [hunk 1]
```diff
@@ -180,8 +181,13 @@ def start_gcloud_token_update_thread(self):
...
         # Wait till GcloudTokenUpdateThread get tokens and dump them to gcloud_token_path
-        while not os.path.exists(self.gcloud_token_path):
-            time.sleep(0.1)
+
+        @timeout(timeout=300)
+        def wait_for_token_file():
+            if not os.path.exists(self.gcloud_token_path):
+                raise RuntimeError('Gcloud token did not appear')
+
+        wait_for_token_file()
```
Comment (bentsi) [inline sdcm/cluster_k8s/gke.py:190]: use `sdcm.wait.wait_for`

Comment (dkropachev) [reply sdcm/cluster_k8s/gke.py]: done

File: sdcm/cluster_k8s/gke.py [hunk 2]
```diff
@@ -179,9 +180,13 @@ def gcloud_token_path(self):
...
         self._gcloud_token_thread.start()
+
+        def wait_for_token_file():
+            if not os.path.exists(self.gcloud_token_path):
+                raise RuntimeError('Gcloud token did not appear')
+
         # Wait till GcloudTokenUpdateThread get tokens and dump them to gcloud_token_path
-        while not os.path.exists(self.gcloud_token_path):
-            time.sleep(0.1)
+        wait_for(wait_for_token_file, timeout=300, step=1, text="Wait for gcloud token", throw_exc=True)
```
Comment (bentsi) [inline sdcm/cluster_k8s/gke.py:189]: better:
```
wait_for(os.path.exists, timeout=30, step=10, text="Wait for gcloud token", throw_exc=True, path=self.gcloud_token_path)
```

Comment (dkropachev) [reply sdcm/cluster_k8s/gke.py]: done

**Review by bentsi — APPROVED** [review]

**Pull request merged**